jps/model-api/src/org/jetbrains/jps/model/java/JavaResourceRootType.java

```diff
@@ -24,10 +24,17 @@
  */
 public class JavaResourceRootType extends JpsElementTypeBase<JavaResourceRootProperties> implements
                                                                                          JpsModuleSourceRootType<JavaResourceRootProperties> {
-  public static final JavaResourceRootType RESOURCE = new JavaResourceRootType();
-  public static final JavaResourceRootType TEST_RESOURCE = new JavaResourceRootType();
+  public static final JavaResourceRootType RESOURCE = new JavaResourceRootType(false);
+  public static final JavaResourceRootType TEST_RESOURCE = new JavaResourceRootType(true);
 
-  private JavaResourceRootType() {
+  private final boolean isForTests;
+  private JavaResourceRootType(boolean isForTests) {
+    this.isForTests = isForTests;
```

Comment:
We use `my` prefix for instance fields in Java classes to avoid such `this.` qualifiers.